### PR TITLE
Security severity for s360 and Update serviceid details

### PR DESCRIPTION
**Pull request by arvindsingh-cloud**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 [hunk 1]
```diff
@@ -418,11 +426,20 @@ class AutoBugLog {
...
         if ($state -eq "Resolved") {
-            $url = "https://dev.azure.com/{0}/{1}/_apis/wit/workitems/{2}?api-version=5.1" -f $this.OrganizationName, $ProjectName, $id
-            $BugTemplate = [ConfigurationManager]::LoadServerConfigFile("TemplateForResolvedBug.json")
-            $BugTemplate = $BugTemplate | ConvertTo-Json -Depth 10 
-            $BugTemplate = $BugTemplate.Replace("{0}", $AssignedTo)           
+            $BugTemplate = $null;
+            #Check if serviceid is not null and current resource scanned serviceid and bug respons serviceid is not equal, then update the service data.
+            if ($this.ShowBugsInS360 -and $serviceId -and ($serviceIdInLoggedBug -ne $serviceId)) 
+            {
+                $BugTemplate = $this.S360BugTemplate($serviceId, $control.ControlItem.ControlSeverity, $true, $AssignedTo)
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1:435]: function rename

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 [hunk 2]
```diff
@@ -451,9 +468,56 @@ class AutoBugLog {
...
+                catch {
+                    Write-Host "Could not update service tree details."
+                }
+            }
         }
-    
+    }
+
+    #status has value if it is called from resolved to activate bug, else the value is empty, if status not needed to change
+    hidden [object] S360BugTemplate($serviceId, $controlSeverity, $status, $assignedTo)
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1:490]: $status -> $reactivate

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 [hunk 3]
```diff
@@ -451,9 +468,56 @@ class AutoBugLog {
...
+        if ($status) {
+            $BugTemplate = $BugTemplate | ConvertTo-Json -Depth 10 
+            $BugTemplate = $BugTemplate.Replace("{0}", "Active")           
+            $BugTemplate = $BugTemplate.Replace("{1}", $AssignedTo)           
+        }
+        else {
+            $BugTemplate = $BugTemplate | Where {$_.path -ne "/fields/System.State" -and $_.path -ne "/fields/System.AssignedTo" }
+            $BugTemplate = $BugTemplate | ConvertTo-Json -Depth 10
+        }
+        $secSeverity = "";
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1:503]: add comment

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 [hunk 4]
```diff
@@ -451,9 +468,56 @@ class AutoBugLog {
...
+            {
+                $BugTemplate = $null;
+                $header = [WebRequestHelper]::GetAuthHeaderFromUriPatch($url)                
+                try 
+                {
+                    $BugTemplate = $this.S360BugTemplate($serviceId, $control.ControlItem.ControlSeverity, "", ""); 
+                    $responseObj = Invoke-RestMethod -Uri $url -Method Patch  -ContentType "application/json-patch+json ; charset=utf-8" -Headers $header -Body $BugTemplate
+                }
+                catch {
+                    Write-Host "Could not update service tree details."
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1:483]: in the bug.

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 (hunk 2 above)
Comment (abhaydaga) [reply src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1:490]: $reactivateBug

Comment (abhaydaga) [reply src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1:490]: UpdateBugSTDetails

Comment (abhaydaga) [reply src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1:490]: UpdateSTBugTemplate

File: src/AzSK.ADO/SVT/SVT.ps1 [hunk 5]
```diff
@@ -213,6 +213,11 @@ function Get-AzSKADOSecurityStatus
 		[Alias("ipt")]
 		$IterationPath,
 
+		[string]
+		[Parameter(Mandatory=$false)]
```
Comment (abhaydaga) [inline src/AzSK.ADO/SVT/SVT.ps1:217]: Security severity of bugs to be logged.

Comment (abhaydaga) [reply src/AzSK.ADO/SVT/SVT.ps1:217]: Specify the security severity of bugs to be logged.

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 (hunk 1 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1]: reanmed

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 (hunk 2 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1]: updated

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 (hunk 3 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1:505]: comment added

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 (hunk 4 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1]: updated

File: src/AzSK.ADO/SVT/SVT.ps1 (hunk 5 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/SVT/SVT.ps1]: added

**Review by abhaydaga — APPROVED** [review]

**Pull request merged**